pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go

```diff
@@ -22,17 +37,156 @@ func init() {
 	sql.AddPlanHook("create logical replication stream", createLogicalReplicationStreamPlanHook, createLogicalReplicationStreamTypeCheck)
 }
 
+var streamCreationHeader = colinfo.ResultColumns{
+	{Name: "job_id", Typ: types.Int},
+}
+
 func createLogicalReplicationStreamPlanHook(
-	ctx context.Context, stmt tree.Statement, p sql.PlanHookState,
+	ctx context.Context, untypedStmt tree.Statement, p sql.PlanHookState,
 ) (sql.PlanHookRowFn, colinfo.ResultColumns, []sql.PlanNode, bool, error) {
-	if _, ok := stmt.(*tree.CreateLogicalReplicationStream); !ok {
+	stmt, ok := untypedStmt.(*tree.CreateLogicalReplicationStream)
+	if !ok {
 		return nil, nil, nil, false, nil
 	}
-	return nil, nil, nil, false, errors.UnimplementedError(issuelink.IssueLink{Detail: "plan hook to be implemented"}, "plan hook to be implemented")
+
+	exprEval := p.ExprEvaluator("LOGICAL REPLICATION STREAM")
+
+	from, err := exprEval.String(ctx, stmt.PGURL)
+	if err != nil {
+		return nil, nil, nil, false, err
+	}
+
+	fn := func(ctx context.Context, _ []sql.PlanNode, resultsCh chan<- tree.Datums) (err error) {
+		defer func() {
+			if err == nil {
+				telemetry.Count("logical_replication_stream.started")
+			}
+		}()
+		ctx, span := tracing.ChildSpan(ctx, stmt.StatementTag())
+		defer span.Finish()
+
+		if !p.ExecCfg().Settings.Version.IsActive(ctx, clusterversion.V24_2) {
+			return pgerror.New(pgcode.FeatureNotSupported,
+				"replication job not supported before V24.2")
+		}
+
+		if err := utilccl.CheckEnterpriseEnabled(
+			p.ExecCfg().Settings,
+			"CREATE LOGICAL REPLICATION",
+		); err != nil {
+			return err
+		}
+
+		if !stmt.Options.IsDefault() {
+			return errors.UnimplementedErrorf(issuelink.IssueLink{}, "logical replication stream options are not yet supported")
+		}
+		if stmt.On.Database != "" {
+			return errors.UnimplementedErrorf(issuelink.IssueLink{}, "logical replication streams on databases are unsupported")
+		}
+		if len(stmt.On.Tables) != len(stmt.Into.Tables) {
+			return pgerror.New(pgcode.InvalidParameterValue, "the same number of source and destination tables must be specified")
+		}
+
+		streamAddress := crosscluster.StreamAddress(from)
+		streamURL, err := streamAddress.URL()
+		if err != nil {
+			return err
+		}
+		streamAddress = crosscluster.StreamAddress(streamURL.String())
+
+		client, err := streamclient.NewStreamClient(ctx, streamAddress, p.ExecCfg().InternalDB)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			_ = client.Close(ctx)
+		}()
+
+		if err := client.Dial(ctx); err != nil {
+			return err
+		}
```

Comment:
Should we do all of the local work before connecting to the remote cluster?

Reply:
Done.